src/components/TodoItem.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable react/require-default-props */
+/* eslint-disable react/destructuring-assignment */
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import styles from './TodoItem.module.css';
+
+const TodoItem = (props) => {
+  const [editing, setEditing] = useState(false);
+  const [editedData, setEditingData] = useState('');
+  const [id, setId] = useState(0);
+
+  const viewMode = {};
+  const editMode = {};
+  if (editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  const handleEditing = () => {
+    const { title, id } = props.checked;
+    setEditing(true);
+    setEditingData(title);
+    setId(id);
+  };
+
+  const handleUpdatedDone = (e) => {
+    if (e.key === 'Enter') {
+      setEditing(false);
+    }
+  };
+
+  const handleChange = (e) => {
+    setEditingData(e.target.value);
+    props.setUpdate(e.target.value, id);
+  };
+  return (
+    <ul>
+      <li>
+        <div onDoubleClick={handleEditing} style={{ viewMode }}>
+          <input type="checkbox" onChange={() => { props.handleChangeProps(props.checked.id); }} />
+          {props.title}
+          {' '}
+          <button type="button" onClick={() => props.deleteTodoProps(props.checked.id)}>Delete</button>
```

Comment:
For more clean code and for best practices, it's better to use object destructuring
ex: `{deleteTodoProps} = props`